src/sentry/toolbar/iframe_view.py

```diff
@@ -0,0 +1,17 @@
+from django.http import HttpRequest, HttpResponse
+from django.utils.decorators import method_decorator
+from django.views.decorators.csrf import csrf_exempt
+from django.views.generic import View
+
+from sentry.web.helpers import render_to_response
+
+
+class IframeView(View):
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request: HttpRequest) -> HttpResponse:
+        if request.method != "GET":
+            return HttpResponse(status=405)
+
+        response = render_to_response("sentry/toolbar/iframe.html")
+        response["X-Frame-Options"] = "ALLOWALL"
```

Comment:
This could be a good place to hook in the organization's allowed domain list.

Reply:
Yes! We're going to be able to use the referer header to know the origin that requested the iframe, and then will check that against an allowlist, also make the header more specific in case it's cached, and there'll be a postMessage call in the template also scoped to the allowed domain.